sdcm/utils/benchmarks.py

```diff
@@ -0,0 +1,137 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2021 ScyllaDB
+import logging
+import re
+
+from sdcm.remote import RemoteCmdRunnerBase, shell_script_cmd
+from sdcm.utils.metaclasses import Singleton
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ScyllaClusterBenchmarkManager(metaclass=Singleton):
+    def __init__(self):
+        """
+        This class is meant for gathering the benchmark results
+        of all the relevant nodes in the cluster and presenting
+        them in unified fashion.
+        Optionally, these can also be committed to remote
+        storage (S3? Argus?)
+        """
+        self._nodes: list["BaseNode"] = []
+        self._benchmark_runners: list[ScyllaNodeBenchmarkRunner] = []
+
+    def add_node(self, new_node: "BaseNode"):
+        if new_node.distro.is_debian_like:
+            self._benchmark_runners.append(ScyllaNodeBenchmarkRunner(new_node))
+        else:
+            LOGGER.info("Skipped installing benchmarking tools on a non-debian-like distro.")
+
+    def install_benchmark_tools(self):
+        for runner in self._benchmark_runners:
+            runner.install_benchmark_tools()
+
+    def run_benchmarks(self):
+        for runner in self._benchmark_runners:
+            runner.run_benchmarks()
+
+    @staticmethod
+    def _collect_benchmark_output():
+        ...
+
+
+class ScyllaNodeBenchmarkRunner:
+    def __init__(self, node: "BaseNode"):
+        """
+        This class is meant to install and run benchmarking
+        tools on given cluster nodes and collect the output.
+        """
+        self._node = node
+        self._remoter: RemoteCmdRunnerBase = node.remoter
+
+    def install_benchmark_tools(self) -> bool:
+        self._clone_sysbench_repo()
+        self._install_ubuntu_prerequisites()
+        self._build_and_install_sysbench()
+        return self._check_sysbench_installation()
+
+    def _install_ubuntu_prerequisites(self) -> bool:
+        package_list = ["make", "automake", "libtool", "pkg-config", "libaio-dev"]
+        try:
+            LOGGER.info("Installing Ubuntu prerequisites for the node benchmarks...")
+            for pkg in package_list:
+                self._node.install_package(pkg)
+            return True
+        except Exception:
+            LOGGER.warning("Failed to install Ubuntu prerequisites for the node benchmarking tools.")
```

Comment:
Why is it "warning" log message knowing that the next step is to "raise" an exception?

Reply:
Fixed (removed the exception reraising)